workitem/workitem_repository.go

```diff
@@ -553,13 +554,27 @@ func ConvertWorkItemStorageToModel(wiType *WorkItemType, wi *WorkItemStorage) (*
 
 // extracted this function from List() in order to close the rows object with "defer" for more readability
 // workaround for https://github.com/lib/pq/issues/81
-func (r *GormWorkItemRepository) listItemsFromDB(ctx context.Context, spaceID uuid.UUID, criteria criteria.Expression, start *int, limit *int) ([]WorkItemStorage, uint64, error) {
+func (r *GormWorkItemRepository) listItemsFromDB(ctx context.Context, spaceID uuid.UUID, criteria criteria.Expression, parentExists *bool, start *int, limit *int) ([]WorkItemStorage, uint64, error) {
 	where, parameters, compileError := Compile(criteria)
 	if compileError != nil {
 		return nil, 0, errors.NewBadParameterError("expression", criteria)
 	}
 	where = where + " AND space_id = ?"
 	parameters = append(parameters, spaceID)
+	pe := true
+	if parentExists == nil {
+		parentExists = &pe
+	}
+	if *parentExists == false {
+		where = where + " AND " + `
+			id not in (
+				SELECT target_id FROM work_item_links
```

Comment:
`EXISTS` subquery is to check whether at least one record is returned, and based on that it will be true or false. But here `IN` query is required to check the value really matches or not. Please correct me if I am wrong.